plugins/adobe-analytics-compatible-version/plugin.test.ts

```diff
@@ -0,0 +1,168 @@
+    },
+    ACPExtensionEventSource: 'com.adobe.eventsource.sharedstate',
+    ACPExtensionEventType: 'com.adobe.eventtype.hub',
+    metadata: {
+      'state.data': {
+        extensions: {
+          'com.adobe.assurance': {
+            version: '1.0.0'
+          },
+          'com.adobe.module.analytics': {
+            version: '1.0.0'
+          }
+        },
+        version: '1.0.0'
+      }
+    }
+  }
+};
+
+const iosClientInfo: ClientInfoIOS = {
+  clientId: '1',
+  timestamp: 10000,
+  type: 'client',
+  uuid: '1',
+  payload: {
+    type: 'connect',
+    deviceInfo: {
+      'Canonical platform name': 'iOS',
+      'Device name': 'My Device',
+      Model: 'IPhone X'
+    }
+  }
+};
+
+const iosClientInfoTwo: ClientInfoIOS = {
+  clientId: '2',
+  timestamp: 10000,
+  type: 'client',
+  uuid: '2',
+  payload: {
+    type: 'connect',
+    deviceInfo: {
+      'Canonical platform name': 'iOS',
+      'Device name': 'My Device',
+      Model: 'IPhone X'
+    }
+  }
+};
+
+const androidClientInfo: ClientInfoAndroid = {
+  clientId: '1',
+  timestamp: 10000,
+  type: 'client',
+  uuid: '1',
+  payload: {
+    deviceInfo: {
+      'Canonical platform name': 'Android'
+    }
+  }
+};
+
+describe('Adobe Analytics Compatible Version', () => {
+  it('should successfully validate for IOS when versions are compatible', () => {
+    const result = plugin([versionEvent, iosClientInfo, iosClientInfoTwo]);
+
+    expect(result).toMatchObject({
+      events: [],
+      message:
+        'The installed Assurance SDK and Adobe Analytics Extensions are compatible.',
+      result: 'matched'
+    });
+  });
+
+  it('should return a validation error when IOS versions are not correct', () => {
+    const result = plugin([invalidVersionEvent, iosClientInfo]);
+
+    expect(result).toMatchObject({
+      events: ['1'],
+      message:
+        'Assurance SDK Version 1.0.0 and Adobe Analytics Extension version 1.0.0 are not compatible!',
+      result: 'not matched'
+    });
+  });
+
+  it('should successfully validate for Android when versions are compatible', () => {
+    const result = plugin([versionEvent, androidClientInfo]);
+
+    expect(result).toMatchObject({
+      events: [],
+      message:
+        'The installed Assurance SDK and Adobe Analytics Extensions are compatible.',
+      result: 'matched'
+    });
+  });
+
+  it('should return a validation error when Android versions are not correct', () => {
+    const result = plugin([invalidVersionEvent, androidClientInfo]);
+
+    expect(result).toMatchObject({
+      events: ['1'],
+      message:
+        'Assurance SDK Version 1.0.0 and Adobe Analytics Extension version 1.0.0 are not compatible!',
+      result: 'not matched'
+    });
+  });
+
+  it('should return an invalid response when no version events are found', () => {
+    const result = plugin([configurationEvent]);
+
+    expect(result).toMatchObject({
+      events: [],
+      message:
```

Comment:
I think we should leave out `message` in the check that way the tests don't have to be adjusted when we need to correct the spelling or grammar of the message or end up localizing these.